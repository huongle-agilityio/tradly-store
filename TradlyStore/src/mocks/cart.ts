import { Cart } from '@/interfaces';

export const CART_QUANTITY = [
  {
    value: '1',
    label: '1',
  },
  {
    value: '2',
    label: '2',
  },
  {
    value: '3',
    label: '3',
  },
  {
    value: '4',
    label: '4',
  },
  {
    value: '5',
    label: '5',
  },
  {
    value: '6',
    label: '6',
  },
  {
    value: '7',
    label: '7',
  },
  {
    value: '8',
    label: '8',
  },
  {
    value: '9',
    label: '9',
  },
  {
    value: '10',
    label: '10',
  },
];

export const CART_lIST: Cart[] = [
  { id: '1', name: 'Item 1', quantity: 2, price: 100, image: '' },
  { id: '2', name: 'Item 2', quantity: 1, price: 50, image: '' },
];
