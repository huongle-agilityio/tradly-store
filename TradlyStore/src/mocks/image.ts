export const IMAGES = 'https://picsum.photos/seed/696/3000/2000';

export const LIST_IMAGES = Array.from({ length: 4 }, () => IMAGES);
