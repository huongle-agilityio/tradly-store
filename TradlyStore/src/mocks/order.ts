// Interfaces
import { OderTracking } from '@/interfaces';

export const ORDER_TRACKING: OderTracking[] = [
  {
    title: 'Order Placed',
    description: 'Order#123455 from Fashion Point',
    date: '05/08/2019',
    time: '11:10 AM',
  },
  {
    title: 'Payment Confirmed',
    description: 'Payment Confirmed Status',
    date: '05/08/2019',
    time: '11:10 AM',
    isActive: true,
  },
  {
    title: 'Processed',
    description: 'Processed Status',
    date: '05/08/2019',
    time: '11:10 AM',
  },
  {
    title: 'Delivered',
    description: 'Delivered Status',
    date: '05/08/2019',
    time: '11:10 AM',
  },
];
