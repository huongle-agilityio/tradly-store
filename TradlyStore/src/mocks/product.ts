export const INT_PRODUCT_CARD = {
  title: '',
  price: 0,
  image: '',
  discount: 0,
  store: {
    name: '',
    image: '',
  },
};

export const ADDITIONAL_DETAILS = [
  {
    label: 'Shipping',
    value: 'shipping',
  },
  {
    label: 'Available',
    value: 'available',
  },
  {
    label: 'Cash on delivery',
    value: 'cashOnDelivery',
  },
];

export const PRODUCT_TEST = [
  {
    id: 1,
    documentId: '1',
    title: 'Grape Soda',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    slideImages: [
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    ],
    quantity: 50,
    price: 34,
    discount: 50,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 2,
    documentId: '1',
    title: 'Tropical Punch',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image:
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    slideImages: [
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    ],
    quantity: 22,
    price: 48,
    discount: 39,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 3,
    documentId: '1',
    title: 'Pineapple Punch',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    slideImages: [
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    ],
    quantity: 20,
    price: 26,
    discount: 29,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 4,
    documentId: '1',
    title: 'Peach Tea',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image:
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    slideImages: [
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    ],
    quantity: 21,
    price: 32,
    discount: 48,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 5,
    documentId: '1',
    title: 'Berry Blast',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    slideImages: [
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    ],
    quantity: 13,
    price: 25,
    discount: 28,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 6,
    documentId: '1',
    title: 'Lemonade',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    slideImages: [
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    ],
    quantity: 19,
    price: 30,
    discount: 27,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 7,
    documentId: '1',
    title: 'Watermelon Soda',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    slideImages: [
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    ],
    quantity: 49,
    price: 40,
    discount: 16,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 8,
    documentId: '1',
    title: 'Mango Juice',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
    slideImages: [
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
    ],
    quantity: 33,
    price: 28,
    discount: 12,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
];
