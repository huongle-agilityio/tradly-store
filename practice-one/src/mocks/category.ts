import { IMAGES } from '@/constants';

export const CATEGORIES = [
  {
    id: 1,
    name: 'Beverages',
    value: 'beverages',
    image: IMAGES.CATEGORY_BEVERAGES,
  },
  {
    id: 2,
    name: 'Bread & Bakery',
    value: 'bread',
    image: IMAGES.CATEGORY_BREAD_BAKERY,
  },
  {
    id: 3,
    name: 'Vegetables',
    value: 'vegetables',
    image: IMAGES.CATEGORY_VEGETABLES,
  },
  {
    id: 4,
    name: 'Fruit',
    value: 'fruit',
    image: IMAGES.CATEGORY_FRUIT,
  },
  {
    id: 5,
    name: 'Egg',
    value: 'egg',
    image: IMAGES.CATEGORY_EGG,
  },
  {
    id: 6,
    name: 'Frozen veg',
    value: 'frozen-veg',
    image: IMAGES.CATEGORY_FROZEN_VEG,
  },
  {
    id: 7,
    name: 'Homecare',
    value: 'homecare',
    image: IMAGES.CATEGORY_HOME_CARE,
  },
  {
    id: 8,
    name: 'Pet Care',
    value: 'pet-care',
    image: IMAGES.CATEGORY_PET_CARE,
  },
];
