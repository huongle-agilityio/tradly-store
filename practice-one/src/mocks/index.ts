export * from './cart';
export * from './product';
export * from './category';
export * from './onboarding';
export * from './order';
export * from './user';
