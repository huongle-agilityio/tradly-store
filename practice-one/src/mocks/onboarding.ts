import { IMAGE_DETAILS } from '@/constants';

export const ONBOARDING_STEPS_IMAGES = [
  {
    image: IMAGE_DETAILS.ONLINE_SHOPPING.src,
    title: 'Empowering Artisans, Farmers & Micro Business',
    alt: IMAGE_DETAILS.ONLINE_SHOPPING.alt,
  },
  {
    image: IMAGE_DETAILS.DIGITAL_COLLABORATION.src,
    title: 'Connecting NGOs, Social Enterprises with Communities',
    alt: IMAGE_DETAILS.DIGITAL_COLLABORATION.alt,
  },
  {
    image: IMAGE_DETAILS.TECHNICAL_ANALYSIS.src,
    title: 'Donate, Invest & Support infrastructure projects',
    alt: IMAGE_DETAILS.TECHNICAL_ANALYSIS.alt,
  },
];
