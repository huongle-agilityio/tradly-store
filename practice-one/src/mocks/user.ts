import { User } from '@/interfaces';

export const USER: User = {
  id: '123',
  email: 'test@example.com',
  name: 'John Doe',
  avatar: '',
  username: 'johndoe',
  phone: '123456789',
};
