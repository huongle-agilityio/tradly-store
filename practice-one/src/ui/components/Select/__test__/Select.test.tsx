import { render, fireEvent, screen } from '@testing-library/react-native';

// Components
import { Select } from '..';

// Mocks
import { CART_QUANTITY } from '@/mocks';

describe('Select Component', () => {
  it('Should renders correctly with placeholder', () => {
    const { toJSON } = render(
      <Select
        value=""
        options={CART_QUANTITY}
        placeholder="Select an option"
        onValueChange={jest.fn()}
      />,
    );

    expect(screen.getByPlaceholderText('Select an option')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should opens options when clicking the input', () => {
    render(
      <Select value="" options={CART_QUANTITY} onValueChange={jest.fn()} />,
    );

    const input = screen.getByTestId('select-box');
    fireEvent.press(input);

    expect(screen.getByText(CART_QUANTITY[0].value)).toBeTruthy();
    expect(screen.getByText(CART_QUANTITY[1].value)).toBeTruthy();
  });

  it('Should closes dropdown when selecting an option', () => {
    const onValueChange = jest.fn();
    render(
      <Select value="" options={CART_QUANTITY} onValueChange={onValueChange} />,
    );

    fireEvent.press(screen.getByTestId('select-box'));
    fireEvent.press(screen.getByText(CART_QUANTITY[0].value));

    expect(onValueChange).toHaveBeenCalledWith(CART_QUANTITY[0].value);
    expect(screen.queryByText(CART_QUANTITY[0].value)).toBeNull();
  });

  it('Should shows error message when error prop is provided', () => {
    render(
      <Select
        value=""
        options={CART_QUANTITY}
        error="This field is required"
        onValueChange={jest.fn()}
      />,
    );

    expect(screen.getByText('This field is required')).toBeTruthy();
  });

  it('Should not open options when disabled', () => {
    render(
      <Select
        value=""
        options={CART_QUANTITY}
        disabled
        onValueChange={jest.fn()}
      />,
    );

    fireEvent.press(screen.getByTestId('select-box'));

    expect(screen.queryByText(CART_QUANTITY[0].value)).toBeNull();
  });

  it('Should not open options when isLoading is true', () => {
    render(
      <Select
        value=""
        options={CART_QUANTITY}
        isLoading
        onValueChange={jest.fn()}
      />,
    );

    fireEvent.press(screen.getByTestId('select-box'));

    expect(screen.queryByText(CART_QUANTITY[0].value)).toBeNull();
  });
});
