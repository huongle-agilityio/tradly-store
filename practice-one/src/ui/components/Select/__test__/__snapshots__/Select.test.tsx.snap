// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component Should renders correctly with placeholder 1`] = `
<View
  style={
    [
      {
        "position": "relative",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "padding": 12,
      }
    }
    testID="select-box"
  >
    <TextInput
      editable={false}
      placeholder="Select an option"
      style={
        {
          "color": "#4f4f4f",
          "fontWeight": "500",
          "lineHeight": 14,
        }
      }
      value=""
    />
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={4}
      bbWidth={9}
      color="#4f4f4f"
      focusable={false}
      height={4}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 4,
            "width": 9,
          },
        ]
      }
      tintColor="#4f4f4f"
      vbHeight={4}
      vbWidth={9}
      width={9}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M8.47396 0C8.65786 0 8.74428 0.227303 8.60683 0.349482L4.63287 3.88189C4.5571 3.94925 4.4429 3.94925 4.36713 3.88189L0.393167 0.349482C0.255716 0.227303 0.342136 0 0.52604 0H8.47396Z"
          fill={
            {
              "type": 2,
            }
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
